services/kuber/server/nodes/nodes.go

```diff
@@ -158,27 +167,79 @@ func (d *Deleter) updateClusterData() {
 
 // deleteFromLonghorn will delete node from nodes.longhorn.io
 // return nil if successful, error otherwise
-func (d *Deleter) deleteFromLonghorn(kc kubectl.Kubectl) error {
-	for _, nodeName := range d.workerNodes {
-		log.Info().Msgf("Deleting node %s from nodes.longhorn.io", nodeName)
-		if err := kc.KubectlDeleteResource("nodes.longhorn.io", nodeName, "longhorn-system"); err != nil {
-			return fmt.Errorf("error while deleting node %s from nodes.longhorn.io : %w", nodeName, err)
+func deleteFromLonghorn(kc kubectl.Kubectl, worker string) error {
+	log.Info().Msgf("Deleting node %s from nodes.longhorn.io", worker)
+	if err := kc.KubectlDeleteResource("nodes.longhorn.io", worker, "-n", longhornNamespace); err != nil {
+		return fmt.Errorf("error while deleting node %s from nodes.longhorn.io : %w", worker, err)
+	}
+	return nil
+}
+
+// assureReplication tries to assure, that replicas for each longhorn volume are migrated to nodes, which will remain in the
+func assureReplication(kc kubectl.Kubectl, worker string) error {
+	// Get replicas and volumes as they can be scheduled on next node, which will be deleted.
+	replicas, err := getReplicas(kc)
+	if err != nil {
+		return fmt.Errorf("error while getting replicas from cluster : %w", err)
+	}
+	volumes, err := getVolumes(kc)
+	if err != nil {
+		return fmt.Errorf("error while getting volumes from cluster  : %w", err)
+	}
+	if reps, ok := replicas[worker]; ok {
+		for _, r := range reps {
+			// Try to force creation of a new replicas from node, which will be deleted.
+			if v, ok := volumes[r.Spec.VolumeName]; ok {
+				// Increase number of replicas in volume.
+				if err := increaseReplicaCount(v, kc); err != nil {
+					return fmt.Errorf("error while increasing number of replicas in volume %s from cluster : %w", v.Metadata.Name, err)
+				}
+				// Wait pvcReplicationTimeout for Longhorn to create new replica.
+				log.Info().Msgf("Waiting %d seconds for new replicas to be scheduled if possible for node %s", pvcReplicationTimeout, worker)
+				time.Sleep(pvcReplicationTimeout * time.Second)
+				// Decrease number of replicas in volume -> original state.
+				if err := revertReplicaCount(v, kc); err != nil {
+					return fmt.Errorf("error while increasing number of replicas in volume %s : %w", v.Metadata.Name, err)
+				}
+				// Delete old replica, on to be deleted node.
```

Comment:
`Delete old replica, on to be deleted node.` -> `Delete old replica on a to-be-deleted node.`